external-crates/move/crates/move-package-alt/src/dependency/external_protocol.rs

```diff
@@ -172,37 +132,35 @@ impl Request {
 
     /// Run the external resolver [resolver] and return its response. The response is guaranteed to
     /// contain the same keys as [self]. This method also forwards the standard error from the
-    /// resolver to this process's standard error
-    pub async fn execute(&self, resolver: &ResolverName) -> anyhow::Result<Response> {
+    /// resolver to `info!`
+    pub async fn execute(&self, resolver: &ResolverName) -> Result<Response, ResolverError> {
         let mut child = Command::new(resolver)
             .arg(RESOLVE_ARG)
             .stdin(Stdio::piped())
             .stdout(Stdio::piped())
             .stderr(Stdio::piped())
             .spawn()
-            .map_err(|e| ResolutionError::io_error(resolver, e))?;
+            .map_err(|e| ResolverError::io_error(resolver, e))?;
 
         send_request(resolver, self, child.stdin.take().expect("stdin exists")).await;
         let response = recv_response(resolver, child).await?;
 
         for key in self.queries.keys() {
             if !response.responses.contains_key(key) {
-                // TODO: failure here is a bad response from the resolver
-                anyhow::bail!("External resolver didn't resolve {key}");
+                return Err(ResolverError::bad_resolver(
+                    resolver,
+                    format!(r#"it didn't resolve "{key}""#),
```

Comment:
Curios, does this work?
```suggestion
                    format!("it didn't resolve \"{key}\""),
```

Alternatively, what I think we do in other parts of the codebase is to put single quotes for things we want to highlight.